app/models/filters/filter_base.rb

```diff
@@ -1123,17 +1123,79 @@ def available_household_types
       }.invert.freeze
     end
 
-    def available_prior_living_situations
-      HUD.living_situations.invert.map do |title, id|
-        [
-          "#{title} (#{id})",
-          id,
-        ]
-      end.to_h
+    def available_prior_living_situations(grouped: false)
+      if grouped
+        {
+          'Homeless' => HUD.homeless_situation_options(as: :prior).map do |id, title|
+            [
+              "#{title} (#{id})",
+              id,
+            ]
+          end.to_h,
+          'Institutional' => HUD.institutional_situation_options(as: :prior).map do |id, title|
+            [
+              "#{title} (#{id})",
+              id,
+            ]
+          end.to_h,
+          'Temporary and Permanent' => HUD.temporary_and_permanent_housing_situation_options(as: :prior).map do |id, title|
+            [
+              "#{title} (#{id})",
+              id,
+            ]
+          end.to_h,
+          'Other' => HUD.other_situation_options(as: :prior).map do |id, title|
+            [
+              "#{title} (#{id})",
+              id,
+            ]
+          end.to_h,
+        }
+      else
+        HUD.living_situations.map do |id, title|
+          [
+            "#{title} (#{id})",
+            id,
+          ]
+        end.to_h
+      end
     end
 
-    def available_destinations
-      HUD.valid_destinations.invert
+    def available_destinations(grouped: false)
+      return HUD.valid_destinations unless grouped
```

Comment:
Should this return this as `"#{title} (#{id})"` to match the grouped case? Or, at least '.invert'? Won't this return the destination ids as the labels?